gateway/gateway.go

```diff
@@ -354,7 +356,9 @@ func (gateway *HandleT) dbWriterWorkerProcess() {
 		})
 		cancel()
 		if err != nil {
-			panic(err)
+			for _, job := range jobList {
+				errorMessagesMap[job.UUID] = err.Error()
+			}
 		}
 		gateway.dbWritesStat.Count(1)
```

Comment:
we cannot be confident that the error will always be `context deadline exceeded`, thus should give precedence to the context's error if there is one.
It would also be fragile to keep adding different errors in the error map, since there will always be a possibility to miss something, e.g. `pq: canceling statement due to user request`
```go
		if err != nil {
			errorMessage := err.Error()
			if ctx.Err() != nil {
				errorMessage = ctx.Err().Error()
			}
			for _, job := range jobList {
				errorMessagesMap[job.UUID] = errorMessage
			}
		}
		cancel()
```

Reply:
any context error would be populated here right?
`errorMessagesMap, err = gateway.jobsDB.StoreWithRetryEachInTx(ctx, tx, jobList)`